src/Fixer/Preload/PreloadExplicitClassSymbolsFixer.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Preload;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Analyzer\ArgumentsAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+final class PreloadExplicitClassSymbolsFixer extends AbstractFixer
+{
+    private $functionsAnalyzer;
+    private $tokenAnalyzer;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->functionsAnalyzer = new FunctionsAnalyzer();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Adds extra `class_exists` before a class to help opcache.preload to discover always-needed symbols.',
+            [
+                new VersionSpecificCodeSample(
+                    '<?php
+use App\Foo;
+
+class MyClass
+{
+    private $foo;
+    public function __construct()
+    {
+        $this->foo = new Foo();
+    }
+}
+',
+                    new VersionSpecification(70300)
+                ),
+            ],
+            'A script for opcache.preload should classes that are used by the average '
+            .'HTTP request. A good preloading script uses reflection to load the '
+            .'hot class and all classes it depends on. Reflection cannot always find '
```

Comment:
what is `hot` class?
there is only ONE hot `class`, or many of them (`classes`)?